Sources/WhisperKit/Core/WhisperKit.swift

```diff
@@ -646,12 +646,25 @@ public class WhisperKit: Transcriber {
                 let decodeWithFallbackStart = Date()
 
                 let tokenSampler = GreedyTokenSampler(temperature: temp, eotToken: tokenizer.endToken, decodingOptions: options)
+                
+                var currentDecodingOptions = options
+                // If language is not passed and usePrefill is false, detect language and set in options
+                if options.language == nil && !options.usePrefillCache{
+                    let languageDecodingResult = try? await textDecoder.detectLanguage(
+                        from: encoderOutput,
+                        using: decoderInputs,
+                        sampler: tokenSampler,
+                        options: options,
+                        temperature: temp
+                    ).first
+                    currentDecodingOptions.language = languageDecodingResult?.language
+                }
 
                 decodingResult = try await textDecoder.decodeText(
                     from: encoderOutput,
                     using: decoderInputs,
                     sampler: tokenSampler,
-                    options: options,
+                    options: currentDecodingOptions,
```

Comment:
Is this renaming unnecessary now?

Reply:
It would be needed as I have to pass the detected language to the decoder loop so it can skip the forward pass on language token.